### PR TITLE
Add linear time shift to dsp.time_shift

### PR DESCRIPTION
**Pull request by mberz**
### Changes proposed in this pull request:

- Allow linear time shifts with different padding types.
- Related to #292 
- Failing test requires #313

### CODE REVIEW

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -1386,6 +1385,26 @@ def time_shift(
...
+        >>> _, axs = plt.subplots(2, 1)
+        >>> pf.plot.time(impulse, ax=axs[0])
+        >>> pf.plot.time(shifted, ax=axs[1])
+        >>> axs[0].set_title('Original signals')
+        >>> axs[1].set_title('Shifted signals')
+        >>> plt.tight_layout()
+
     """
     shift = np.atleast_1d(shift)
     if shift.size == 1:
```
Comment (f-brinkmann) [inline pyfar/dsp/dsp.py:1410]: wouldn't it be more flexible to broadcast to shift to signal.cshape?

Comment (mberz) [reply pyfar/dsp/dsp.py]: Functionality is the same, but you solution is more compact.

**Comment by mberz** [comment]
Merge issues should be resolved now

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
- Updated the docstring and fixed a bug in the new plot example.
- I think the default for pad_value must be changed and tests for the linear shift should be added

**Review by sikersten — APPROVED** [review]

**Pull request merged**